src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostedService.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public abstract class LinuxContainerInitializationHostedService : IHostedService
+    {
+        private readonly IEnvironment _environment;
+        private readonly IInstanceManager _instanceManager;
+        private readonly ILogger _logger;
+        private readonly StartupContextProvider _startupContextProvider;
+        private CancellationToken _cancellationToken;
+
+        public LinuxContainerInitializationHostedService(IEnvironment environment, IInstanceManager instanceManager, ILogger<LinuxContainerInitializationHostedService> logger, StartupContextProvider startupContextProvider)
+        {
+            _environment = environment;
+            _instanceManager = instanceManager;
+            _logger = logger;
+            _startupContextProvider = startupContextProvider;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Initializing LinuxContainerInitializationService.");
```

Comment:
Recommend removing type specifics, and just change this log message to "Starting container initialization service.". The logger will contain the subclass details

Reply:
Changed 👍